docs/docs/visual-testing-with-storybook.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Visual Testing with Storybook
+---
+
+Knowing your components look as intended in every permutation is not only a great way to test them visually, but also provides "living documentation" for them. This makes it easier for teams to 1) know what components are available to them in a given project and 2) what props those components accept and what all of the states of that component are. As your project grows over time having this information available will be invaluable.
+
+## Setting up your environment
+
+In order to set up Storybook you will need to install a few dependencies and then do some custom configuration to make Storybook work with your Gatsby environment. The first thing is to get the Storybook CLI installed if you don't have it already.
+
+```sh
+npm install -g @storybook/cli
+```
+
+Once the CLI is installed, the next step is to run the `getstorybook` command that is now available from the root directory of your Gatsby project.
+
+```sh
+cd my-awesome-gatsby-project
+getstorybook
+```
+
+The `getstorybook` command will then bootstrap the basic config necessary to run Storybook for a React project. However, since this is for a Gatsby project, we need to update the default Storybook configuration a bit so that we don't get errors when trying to use Gatsby specific components inside of the stories.
+
+To update your Storybook config open `.storybook/config.js` and add the following before the `configure` method at the bottom of the file.
+
+```js
+// Gatsby's Link overrides:
+// Gatsby defines a global called ___loader to prevent its method calls from creating console errors we override it here
+global.___loader = {
+  enqueue: () => {},
+  hovering: () => {},
+}
+
+// Gatsby internal mocking to prevent unnecessary errors in storybook testing environment
+global.__PATH_PREFIX__ = ""
+
+// This is to utilized to override the window.___navigate method Gatsby defines and uses to report what path a Link would be taking us to if it wasn't inside a storybook
+window.___navigate = pathname => {
+  action("NavigateTo:")(pathname)
+}
+```
+
+Next we need up make some adjustments to Storybook's default `webpack` configuration so that we can tranpile Gatsby source files, as well as ensure we have all the necessary `babel` plugins we need to transpile Gatsby components.
+
+Create a new file called `webpack.config.js` in the `.storybook` folder created by the Storybook CLI. Then place the following in that file.
+
+```js
+module.exports = (baseConfig, env, defaultConfig) => {
+  // Transpile Gatsby module because Gastby includes un-transpiled ES6 code.
```

Comment:
We have an open issue for this. Maybe I'll fix this today and add a transpiled `main` field?

See #8821

Reply:
Once #9123 is merged, we can remove this 🎉

Reply:
Fixed!

Reply:
#9123 did fix my situation where i was unit testing in a context that needed common-js, but i'm still seeing raw jsx in the `module` defined by the `gatsby` package, which [causes problems](https://travis-ci.org/dsmjs/components/jobs/442698875#L730-L744) for my storybook instance that was working before upgrading to v2. 

normally `module` is still transpiled to some degree so that that it is at least valid js (which jsx is not) at some lowest common denominator, but still uses es module `import`/`export`. i normally handle producing es-module for `module` and common-js for `main` [using rollup](https://github.com/dsmjs/components/blob/master/rollup.config.js). if this were in place, there shouldn't be a need to transpile the `gatsby` package further when depending on it in a storybook project, etc.

if the team would be open to a build step that would produce both `main` and `module` with a process similar to this, i'd be happy to contribute. thoughts?